sdk/keyvault/azure-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/CryptographyClientBuilder.java

```diff
@@ -0,0 +1,248 @@
+     * either of ({@link CryptographyClientBuilder#keyIdentifier(String)} or {@link CryptographyClientBuilder#jsonWebKey(JsonWebKey)} have not been set.
+     */
+    public CryptographyAsyncClient buildAsyncClient() {
+        Configuration buildConfiguration = (configuration == null) ? ConfigurationManager.getConfiguration().clone() : configuration;
+
+        if (jsonWebKey == null && Strings.isNullOrEmpty(keyId)) {
+            throw new IllegalStateException("Json Web Key or jsonWebKey identifier are required to create cryptography client");
+        }
+
+        if (pipeline != null) {
+            if(jsonWebKey != null) {
+                return new CryptographyAsyncClient(jsonWebKey, pipeline);
+            } else {
+                return new CryptographyAsyncClient(keyId, pipeline);
+            }
+        }
+
+        if (credential == null) {
+            throw new IllegalStateException(KeyVaultErrorCodeStrings.getErrorString(KeyVaultErrorCodeStrings.CREDENTIAL_REQUIRED));
+        }
+
+        // Closest to API goes first, closest to wire goes last.
+        final List<HttpPipelinePolicy> policies = new ArrayList<>();
+        policies.add(new UserAgentPolicy(AzureKeyVaultConfiguration.SDK_NAME, AzureKeyVaultConfiguration.SDK_VERSION, buildConfiguration));
+        policies.add(retryPolicy);
+        policies.add(new KeyVaultCredentialPolicy(credential));
+        policies.addAll(this.policies);
+        policies.add(new HttpLoggingPolicy(httpLogDetailLevel));
+
+        HttpPipeline pipeline = HttpPipeline.builder()
+                .policies(policies.toArray(new HttpPipelinePolicy[0]))
+                .httpClient(httpClient)
+                .build();
+
+        if(jsonWebKey != null) {
+            return new CryptographyAsyncClient(jsonWebKey, pipeline);
+        } else {
+            return new CryptographyAsyncClient(keyId, pipeline);
+        }
+    }
+
+    /**
+     * Sets the identifier of the jsonWebKey from Azure Key Vault to be used for cryptography operations.
+     *
+     * @param keyId The jsonWebKey identifier representing the jsonWebKey stored in jsonWebKey vault.
+     * @return the updated {@link CryptographyClientBuilder} object.
+     * @throws IllegalArgumentException if {@code endpoint} is null or it cannot be parsed into a valid URL.
+     */
+    public CryptographyClientBuilder keyIdentifier(String keyId) {
+        this.keyId = keyId;
+        return this;
+    }
+
+    /**
+     * Sets the identifier of the jsonWebKey from Azure Key Vault to be used for cryptography operations.
+     *
+     * @param jsonWebKey The Json web jsonWebKey to be used for cryptography operations.
+     * @return the updated {@link CryptographyClientBuilder} object.
+     * @throws IllegalArgumentException if {@code endpoint} is null or it cannot be parsed into a valid URL.
+     */
+    public CryptographyClientBuilder jsonWebKey(JsonWebKey jsonWebKey) {
```

Comment:
Are JsonWebKey and KeyId allowed to be used at the same time? If not we may want to null out the other when on is set?

Reply:
Not null out, but I've mentioned in javadocs JWK takes precedence over Key Id.